nimbus/src/persistence.rs

```diff
@@ -321,6 +343,29 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_corrupt_db() -> Result<()> {
+        let path = "test_corrupt_db";
+        let tmp_dir = TempDir::new(path)?;
+
+        let db_dir = tmp_dir.path().join("db");
+        fs::create_dir(db_dir.clone())?;
+
+        // The database filename differs depending on the rkv mode.
+        #[cfg(feature = "rkv-safe-mode")]
+        let db_file = db_dir.join("data.safe.bin");
+        #[cfg(not(feature = "rkv-safe-mode"))]
+        let db_file = db_dir.join("data.mdb");
+
+        fs::write(&db_file, b"Not a database!")?;
+        assert_eq!(fs::metadata(&db_file)?.len(), 15);
+        // Opening the DB should delete the corrupt file and replace it.
+        Database::new(&tmp_dir)?;
+        // Old contents should be removed and replaced with actual data.
+        assert_ne!(fs::metadata(&db_file)?.len(), 15);
```

Comment:
What is the significance of `15` here?

Why would `fs::metadata(&db_file)?` be `b"Not a database!"` in the first instance?

Is this `assert_ne!` proving that we've now got a real database?

Not saying it's wrong, I'm just confused.

Reply:
> What is the significance of `15` here?

That's the length of "Not a database!" - I'll make that more explicit.

> Why would `fs::metadata(&db_file)?` be `b"Not a database!"` in the first instance?

Line 361 writes that - so we expect to see that until the DB has been deleted and recreated.

> Is this `assert_ne!` proving that we've now got a real database?

Yes - I'm not sure how to otherwise check that - I guess I could just check rkv can open it, but that seems easier and just as effective.

Reply:
FTR, I stuck with the same strategy but replaced the magic number with a `.len()` call